multitenancy-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/multitenancy/autoconfig/MultiTenantEventProcessorControlService.java

```diff
@@ -211,7 +207,14 @@ private Optional<String> tokenStoreIdentifierFor(String processorName) {
     }
 
     private static String processorNameFromCombination(String processorAndContext) {
-        return processorAndContext.substring(0, processorAndContext.indexOf("@"));
+        int index = processorAndContext.indexOf("@");
+        return index == -1 ? processorAndContext : processorAndContext.substring(0, index);
+    }
+
+    private static String contextFromCombination(String processorAndContext) {
+        int index = processorAndContext.indexOf("@");
+        //if there is no context name in the processorAndContext, return the _admin as default
+        return index == -1 ? "_admin" : processorAndContext.substring(index + 1);
```

Comment:
I can't really oversee what the outcome is of setting the returned context to `_admin` here if the given `processorAndContext` doesn't have an `@` in it. Could you explain to me why you've chosen this default?

Reply:
In non-multitenant version, Control Service is always connected to default context, even if event processor is configured to listen from external context (eg: multi_stremable message source).

For multi-tenant extension, context is usually part of the processor name, but when not the case, we default to default context, which is "_admin" context, only one that we can assume always exists and is used for other similar operations, like getting connection to all contexts updates which extension also uses